### ISSUE TITLE
print won't print without an argument

### ISSUE THREAD

**Issue by notwa** [kuroko-lang/kuroko#22]
```python
print()  # prints newline in python, nothing in kuroko
print("")  # prints in both

# also...
print("hello", end=" ")
print(end="world")  # this only works in python
print()
print("hello", end=" ")
print("", end="world")  # this works in both
```

the case with `print(end="blah")` not printing anything had me especially confused when I was porting a script, as I have a (bad?) habit of using that form.

this is with Kuroko 1.2.3 (commit 650f324) and [a tiny patch to build with MSYS2](https://gist.github.com/notwa/9ae1cb0d901a64481d52d379f557d9b0) (`MSYSTEM=MSYS`).

**Comment by klange** [comment]
Ah, looks like a pretty straightforward case of just completely neglecting `argc == 0`... I could have sworn this was being handled at some point, but likely before `print` become a normal function.

```diff
diff --git a/src/builtins.c b/src/builtins.c
index ba8a711..ef0f46e 100644
--- a/src/builtins.c
+++ b/src/builtins.c
@@ -512,6 +512,11 @@ KRK_FUNC(print,{
 			endLen = AS_STRING(endVal)->length;
 		}
 	}
+	if (!argc) {
+		for (size_t j = 0; j < endLen; ++j) {
+			fputc(end[j], stdout);
+		}
+	}
 	for (int i = 0; i < argc; ++i) {
 		KrkValue printable = argv[i];
 		if (IS_STRING(printable)) { /* krk_printValue runs repr */
```

As for the MSYS patch, I will take a look, but I suspect a more direct fix would be to change [the bit in the Makefile that's checking for mingw](https://github.com/kuroko-lang/kuroko/blob/master/Makefile#L24) so that MSYS takes the same path. Native Windows builds without cygwin or the like should work well.

**Comment by notwa** [comment]
ah, I wasn't suggesting the patch so much as I was including it for reproducibility.

taking the other branch in the Makefile *almost* works; the pthread library is named differently here:
```diff
-WINLIBS= -l:libwinpthread.a
+WINLIBS= -l:libpthread.a
```
and gcc still warns about ignoring the "hidden" visibility attribute.

also, I have a few other small issues to report relating to built-ins. would you prefer if I made individual github issues like this, or something else?

**Comment by klange** [comment]
> is this supposed to be ifndef _WIN32?

No, that's for [Sortix](https://gitlab.com/sortix/sortix/), which explicitly does not have `putenv`.

**Comment by klange** [comment]
> also, I have a few other small issues to report relating to built-ins. would you prefer if I made individual github issues like this, or something else?

Let's do a separate issue for each function.

**Comment by klange** [comment]
Closing as fixed in 21b459919b98c084a9ab44da4ebc6ad99d6e09dd
